File: src/statesman/commands/SaveCommand.java
package statesman.commands;

import java.io.IOException;

import statesman.Content;
import statesman.Interpreter;

public class SaveCommand implements Command {

    public static final String Id = "save";
    
    public SaveCommand() {
    }

    @Override
    public Command createInstance(String[] arguments) {
        return new SaveCommand();
    }

    @Override
    public void execute() {
        System.out.println(Content.getScript().getMessage("sl_1"));
        String name = Interpreter.getScanner().nextLine().trim().toLowerCase();
        if (name.isBlank()) {
            System.out.println(Content.getScript().getMessage("sl_2"));
            return;
        }
        if (name.length() > 255) {
            System.out.println(Content.getScript().getMessage("sl_3"));
            return;
        }
        boolean gameSaved = false;
        try {
            Content.saveState(name);
            gameSaved = true;
        } catch (IOException e) {
            System.out.println(Content.getScript().getMessage("sl_6"));
        }
        
        if (gameSaved) {
            System.out.println(Content.getScript().getMessage("sl_7"));
        }
    }

}

File: src/Commands/SaveCommand.cs
using System;
using System.IO;

namespace Statesman.Commands
{
    public class SaveCommand : Command
    {
        public static readonly string ID = "save";

        public SaveCommand()
        {
        }

        public override Command CreateInstance(string[] arguments)
        {
            return new SaveCommand();
        }

        public override void Execute()
        {
            Console.WriteLine(Content.Script.FindMessage("sl_1"));
            string name = Console.ReadLine().Trim().ToLowerInvariant();
            if (string.IsNullOrWhiteSpace(name))
            {
                Console.WriteLine(Content.Script.FindMessage("sl_2"));
                return;
            }
            if (name.Length > 255)
            {
                Console.WriteLine(Content.Script.FindMessage("sl_3"));
                return;
            }
            bool gameSaved = false;
            try
            {
                Content.SaveState(name);
                gameSaved = true;
            }
            catch (IOException)
            {
                Console.WriteLine(Content.Script.FindMessage("sl_6"));
            }

            if (gameSaved)
            {
                Console.WriteLine(Content.Script.FindMessage("sl_7"));
            }
        }
    }
}
